proposals/v3/health.md

```diff
@@ -0,0 +1,25 @@
+# Health Checks
+As of Synse API v2, health checks on Synse services are not well-defined.
+While there is some simple notion of health for some of the Synse components,
+not all of them have health checks, and those that do have very simple health
+checks.
+
+For API v3, it would be beneficial to add in more comprehensive health checks
+for all the core Synse components. This will make them easier to manage via
+container orchestration, like Kubernetes.
+
+
+## Synse Server
+Synse Server already exposes an HTTP API, so a `/health` endpoint can be
+added to provide the health check for the service.
+
+> TODO: Figure out what data points can be used to evaluate whether synse
+> server is healthy or not.
+
+
+## Synse Plugins
+Plugins currently expose a GRPC API. It is unclear whether or not the
+SDK should expose a simple HTTP API for health checks, a script is put in
```

Comment:
`/plugins` exposes plugin health via Synse Server, but the plugins themselves don't expose any endpoint to describe their health. The primary purpose of this would be for adding a liveness/readiness probe in kubernetes for the plugins.